### DISCUSSION TITLE
The field in fieldprobe different from dump in diagnostic with openpmd.

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4487] [Q&A] [answered]
I find out that the field getting by fieldprobe in reducefield have some high frequency noise, but the field dump in openpmd in fulldiagnostic don't have. The field of fieldprobe come from the "*Bfield_aux" in "Wrapx" class, does it is the raw data that not have timeaverage in cell but the fulldiagnostic have it?

**Comment by ax3l** [comment, marked as answer]
Hi @zhazhajust,

That is possible, because both diagnostics work slightly different.

The **full diagnostics**:
- center the fields to the same time t (no half-step offset) and position on the cell - the latter is filtering the Nyquist frequency for most fields
- use for j and rho the numerical particle shape of the species in the simulation, which uses a b-spline

The **field probe** does not do any filtering or time-centering, but you can increase the shape of the particle gather as well:
- https://github.com/ECP-WarpX/WarpX/blob/23.12/Source/Diagnostics/ReducedDiags/FieldProbe.cpp#L138
- which is `<diag>.interp_order = ...`

In summary, increase `<diag>.interp_order = ...` for your field probe to filter out more of the high-frequency noise you are seeing.

Ref.: https://warpx.readthedocs.io/en/latest/usage/parameters.html#diagnostics-and-output

**Reply by jiecaix** [reply]
Hi @ax3l,

Thank you very much for the detailed description. 

It's look like a good idea, but I find that the non-physical fields I got was coming from areas close to boundaries, so the ```"<diag>.interp_order"``` seems don't work.

I'm using a refinement level 0 and level 1 in the simulation. The field near the boundary of level 1 to level 0 got by "field probe" have a strong field sharp compared to the field in level 0. But this strong field don't exist in full diagnostics both level 0 and level 1. 

So I wonder if possible to sample the field probe directly use the full diagnostics field or something to remove field distortion of boundaries, or just sample the field in level 0 because it look like totally continues in full diagnostics. 

In refinement simulation, the particle was divided to the 2 level, I try to get the Efield_aux in level 0 with the particle in level 1 to make the field probe only probe the field in level 0, but unfortunately it crashed.
